src/protocols/secure_channel/RendezvousParameters.h

```diff
@@ -63,12 +63,19 @@ class RendezvousParameters
 
     bool HasPeerAddress() const { return mPeerAddress.IsInitialized(); }
     Transport::PeerAddress GetPeerAddress() const { return mPeerAddress; }
+    const Optional<ByteSpan> GetCSRNonce() const {return mCSRNonce;}
     RendezvousParameters & SetPeerAddress(const Transport::PeerAddress & peerAddress)
     {
         mPeerAddress = peerAddress;
         return *this;
     }
 
+    RendezvousParameters & SetCSRNonce(ByteSpan csrNonce)
```

Comment:
This needs _very_ clear lifetime documentation.  The buffer that `csrNonce` points into MUST outlive the `RendezvousParameters` struct.  

I don't see this function actually being used anywhere, not even in tests (should there be tests here?), so hard to say how that will work in practice.

Reply:
So this PR is focussed on just allowing passing the CSRNonce into the commissioning flow. Will have another PR to make the jni layer changes for the Android app to be able to  pass the CSRNonce using the setter.

Reply:
Added the documentation for the csrNonce param